### ISSUE TITLE
Trait version mismatches aren't surfaced in errors

### ISSUE THREAD

**Issue by bbangert** [rust-lang/rust#53795]
When using the sentry 0.8.0 crate with my project using error-chain 0.12.0, an error happened attempting to display the error:

```
error[E0277]: the trait bound `errors::Error: error_chain::ChainedError` is not satisfied
   --> src/client.rs:468:21
    |
468 |                     capture_error_chain(&e);
    |                     ^^^^^^^^^^^^^^^^^^^ the trait `error_chain::ChainedError` is not implemented for `errors::Error`
    |
    = note: required by `sentry::integrations::error_chain::capture_error_chain
```
The errors::Error object did implement the trait, however it was the 0.12 version of the trait, and sentry 0.8.0 was built with error-chain 0.11.0. I got lucky in guessing this was the issue after a few hours of tweaking things and trying to understand how the trait wasn't being satisfied.

Ideally the rust compiler should note the trait version mismatch somehow.

**Comment by estebank** [comment]
CC https://github.com/rust-lang/rust/issues/46388 https://github.com/rust-lang/rust/issues/22750 https://github.com/rust-lang/rust/issues/51458

**Comment by estebank** [comment]
I believe this is fixed by #124944. Without a repro I can't confirm. If this comes up again, please reopen this ticket or open a new one.
